src/appellate/utils.js

```diff
@@ -48,12 +92,53 @@ let APPELLATE = {
 
   // Create a list of doc_ids from the list of all links available on the page
   findDocLinksFromAnchors: (nodeList) => {
-    const links = [];
+    let links = [];
+    let docsToCases = {};
     Array.from(nodeList).map((a) => {
-      if (a.title !== 'Open Document') return;
+      if (!PACER.isDocumentUrl(a.href)) return;
+
+      // this regex will match the doc_id and case_id passed as an argument in the
+      // onclick event of each anchor element related to a document
+      let doDocPost = /^return doDocPostURL\('([^']*)','([^']*)'\);/.exec(a.getAttribute('onclick'));
+      let params = {};
+      if (doDocPost) {
+        [, params.doc_id, params.case_id] = doDocPost;
+        docsToCases[params.doc_id] = params.case_id;
+      }
+
+      a.removeAttribute('onclick');
+      a.setAttribute('target', '_self');
+
+      // clone and replace anchor elements to remove all listeners
+      let clonedNode = a.cloneNode(true);
+      a.replaceWith(clonedNode);
+
+      // add a new listener that allows the anchors to target the active tab
+      clonedNode.addEventListener('click', (event) => {
+        let form = document.getElementsByName('doDocPostURLForm')[0];
+        if (form) {
+          form.dls_id.value = params.doc_id;
+          form.caseId.value = params.case_id;
+          form.submit();
+        }
+      });
```

Comment:
How much of this is still needed if we update the form's `target` attribute so it doesn't open new tabs? I thought this would be unneeded if we did that, and we could just let the normal JS run without creating our own listeners.

Reply:
After changing the form's target attribute, I thought I could remove our listener too but then I tested a new implementation of the extension that used the default JS and noticed that the URL for the attachment page didn't include the **servlet** parameter. 

The problem with the URL that doesn't have the servlet parameter is that we don't have a reliable way to get the **case_id** from the attachment page. We can use the tab storage to retrieve the metadata we've captured and use the **case_id** from the previous page( **General Docket** or **Case Summary**) but I'm sure if an attachment page could have a different **case_id** than the one found in the previous pages ( could the attachment page  belong to another docket but is listed on the one we're reviewing?)

I didn't remove the new listener because I think the **servlet** parameter is important for the **attachment page**. this parameter includes the **doc_id** for this page. We can extract it with a regex expression and then use the mapping ( we create this object parsing the `onclick` event from the document anchor elements and holds the **doc_id** and the **case_id** as a key-value pair) to get the **case_id**. I think this approach is better because we're capturing the **case_id** for each document so we could have different ids inside the mapping.

Reply:
We concluded on Slack that documents can belong to multiple cases and that some cases just reference docs in other cases in their text. So we can't assume that attachment pages are from the same case as the docket that leads to them.